streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorder.java

```diff
@@ -17,117 +17,176 @@
 package org.apache.kafka.streams.state.internals.metrics;
 
 import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.utils.LogContext;
 import org.apache.kafka.streams.processor.TaskId;
 import org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl;
 import org.apache.kafka.streams.state.internals.metrics.RocksDBMetrics.RocksDBMetricContext;
 import org.rocksdb.Statistics;
 import org.rocksdb.StatsLevel;
+import org.rocksdb.TickerType;
+import org.slf4j.Logger;
 
-import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class RocksDBMetricsRecorder {
+    private final Logger logger;
 
     private Sensor bytesWrittenToDatabaseSensor;
-    private Sensor bytesReadToDatabaseSensor;
+    private Sensor bytesReadFromDatabaseSensor;
     private Sensor memtableBytesFlushedSensor;
     private Sensor memtableHitRatioSensor;
-    private Sensor memtableAvgFlushTimeSensor;
-    private Sensor memtableMinFlushTimeSensor;
-    private Sensor memtableMaxFlushTimeSensor;
     private Sensor writeStallDurationSensor;
     private Sensor blockCacheDataHitRatioSensor;
     private Sensor blockCacheIndexHitRatioSensor;
     private Sensor blockCacheFilterHitRatioSensor;
     private Sensor bytesReadDuringCompactionSensor;
     private Sensor bytesWrittenDuringCompactionSensor;
-    private Sensor compactionTimeAvgSensor;
-    private Sensor compactionTimeMinSensor;
-    private Sensor compactionTimeMaxSensor;
     private Sensor numberOfOpenFilesSensor;
     private Sensor numberOfFileErrorsSensor;
 
-    final private Map<String, Statistics> statisticsToRecord = new HashMap<>();
-    final private String metricsScope;
-    final private String storeName;
+    private final Map<String, Statistics> statisticsToRecord = new ConcurrentHashMap<>();
+    private final String metricsScope;
+    private final String storeName;
+    private TaskId taskId;
+    private StreamsMetricsImpl streamsMetrics;
+    private boolean isInitialized = false;
 
     public RocksDBMetricsRecorder(final String metricsScope, final String storeName) {
         this.metricsScope = metricsScope;
         this.storeName = storeName;
+        final LogContext logContext = new LogContext(String.format("[RocksDB Metrics Recorder for %s] ", storeName));
```

Comment:
It's a bit confusing to have two storeNames variable, the first one is actually for the logical storeName (which would be the same to the physical store name if it is kv-store) and the second is for the physical storeName (e.g. for segmented store its name is `storeName-segmentID`).

Maybe rename the second to rocksDBName?

Reply:
Renaming is a good idea. I renamed the physical `storeName` to `segmentName`.